src/TorchSharp/Utils/CRC32C.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation and Contributors.  All Rights Reserved.  See LICENSE in the project root for license information.
+
+// Note: the native implementation of CRC32C has the following copyright / license notice:
+
+/* Part of CRC-32C library: https://crc32c.machinezoo.com/ */
+/*
+  Copyright (c) 2013 - 2014, 2016 Mark Adler, Robert Vazan, Max Vysokikh
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the author be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+  claim that you wrote the original software. If you use this software
+  in a product, an acknowledgment in the product documentation would be
+  appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+  misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace TorchSharp.Utils
+{
+    public static class CRC32C
+    {
+        // uint32_t crc32c_append(uint32_t crc, buffer input, size_t length)
+
+        [DllImport("LibTorchSharp")]
+        extern static uint crc32c_append(uint crc, IntPtr value, ulong length);
```

Comment:
IMHO, it would be better to rewrite the original C file in C#. The differences would be minimal.

Reply:
I'm holding out hope that the HW acceleration code in the original source might be something we can bring back in when computing CRC on larger data blobs, such as images, and such. That'll require C.

Reply:
Looks like simple bit shifting to me. No manual AVX calls or anything like that.